### ISSUE TITLE
`SkyCoord.represent_as` does not give easy access to velocities

### ISSUE THREAD

**Issue by jobovy** [astropy/astropy#12817]
### Description

Using `represent_as` on a `SkyCoord` defined with velocities does not allow one to get the velocities (easily). E.g.,

```
from astropy.coordinates import SkyCoord
import astropy.units as u
c= SkyCoord(ra=20.*u.deg,dec=30.*u.deg,distance=2.*u.kpc,
                       pm_ra_cosdec=-10.*u.mas/u.yr,pm_dec=20.*u.mas/u.yr,
                       radial_velocity=50.*u.km/u.s)
c.transform_to('galactic')
print(c.represent_as('cylindrical').d_phi)
```

fails, while

```
c.representation_type= 'cylindrical'
print(c.d_phi)
```

does work. Note that the information is there in the `represent_as` output as

```
print(c.represent_as('cylindrical')._differentials['s'].d_phi)
```

but I assume that's not how I'm supposed to get the velocities...

### System Details

```
macOS-12.1-arm64-arm-64bit
Python 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:41:07) 
[Clang 11.1.0 ]
Numpy 1.22.0
pyerfa 2.0.0.1
astropy 5.0
Scipy 1.7.3
Matplotlib 3.5.1
```

**Comment by ayshih** [comment]
> but I assume that's not how I'm supposed to get the velocities...

I'd use `differentials['s']` instead of `_differentials['s']` (i.e., no underscore), but that's what you have to do.  `CylindricalRepresentation` is general and allows for non-velocity differentials (i.e., differentials with respect to something other than time), so it doesn't automatically expose velocity components (e.g., `d_phi`) as attributes.  `SkyCoord` basically supports only velocity differentials, so chooses to expose the velocity components as attributes.

**Comment by nstarman** [comment]
SkyCoord does have a built-in way to do this, but it's not obvious as the online documentation doesn't show the ``represent_as`` method...

```python
from astropy.coordinates import SkyCoord
import astropy.units as u
c= SkyCoord(ra=20.*u.deg,dec=30.*u.deg,distance=2.*u.kpc,
                       pm_ra_cosdec=-10.*u.mas/u.yr,pm_dec=20.*u.mas/u.yr,
                       radial_velocity=50.*u.km/u.s)
c.transform_to('galactic')
c.represent_as('cylindrical', s="cylindrical").differentials["s"].d_phi
```

The "s=" isn't strictly required in this example, but one could specify e.g. "spherical" to have cylindrical positions but spherical velocities.

One thing that would be nice is if ``transform_to`` allowed the representation and differential types to be specified, so the following were possible.

```python
c.transform_to('galactic', representation_type="cylindrical", differential_type="cylindrical").d_phi
```

**Comment by jobovy** [comment]
Yes, I know that I can get `d_phi` through the differentials (that was in my original comment; although I don't think this is documented, I just found it by trying). What I don't understand is the inconsistency between
```
c.representation_type= 'cylindrical'
print(c.d_phi)
```
which works and
```
print(c.represent_as('cylindrical').d_phi)
```
which doesn't. I would think these should be equivalent. Changing the representation type is annoying, because then you lose access to the previous attributes, so when you want to mix coordinates things become difficult (e.g., doing a cut on Galactic longitude and vertical height).

More generally, I do not understand why `astropy` makes it so difficult to access the velocities!

**Comment by ayshih** [comment]
The seeming inconsistency between your two blocks is that `SkyCoord.represent_as()` does not return a `SkyCoord`, which is why the behavior is different.  In your first block, you've changed the representation type, but `c` is still a `SkyCoord`.  Thus, it retains the notion that velocity differentials are special and provides direct access to the components as attributes.  In your second code block, `c.represent_as()` is is equivalent to `c.frame.represent_as()` (which is why it is not a documented method of `SkyCoord`), and returns a `CylindricalRepresentation`.  And, as I said in my earlier comment, `CylindricalRepresentation` gives no special status to velocity differentials, so `CylindricalRepresentation.d_phi` doesn't exist.

Changing the return type of `represent_as()` may be too disruptive at this point, but if there's enough interest, perhaps a new method could be created to do what you want (i.e., return a modified `SkyCoord`)?

> Changing the representation type is annoying, because then you lose access to the previous attributes, so when you want to mix coordinates things become difficult (e.g., doing a cut on Galactic longitude and vertical height).

It's of course a bit additionally cumbersome, but you could first create a copy of `c`, and change `representation_type` on that.

**Comment by jobovy** [comment]
> In your second code block, `c.represent_as()` is is equivalent to `c.frame.represent_as()` (which is why it is not a documented method of `SkyCoord`)

It is documented here: https://docs.astropy.org/en/stable/coordinates/skycoord.html (search for 'represent_as'), where it says:

> Note that if all you want is a particular representation without changing the state of the [SkyCoord](https://docs.astropy.org/en/stable/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord) object, you should instead use the astropy.coordinates.SkyCoord.represent_as() method:

thus, I assumed that it would give me all of the attributes that I would get by changing the representation type.

Anyway, if people don't think this should be fixed, feel free to close this. I have much more convenient ways of getting what I want to do (`galpy.orbit.Orbit(c).ANY_COORDINATE_YOU_WANT()`)
